test/libsolidity/util/BytesUtils.cpp

```diff
@@ -327,7 +366,12 @@ string BytesUtils::formatBytesRange(
 	{
 		bytes byteRange{it, it + static_cast<long>(parameter.abiType.size)};
 
-		if (!parameter.matchesBytes(byteRange))
+		if (!parameter.matchesBytes(byteRange) ||
+			// Fixed point values may look exactly the same, but their actual values may be different if
+			// it uses a different fixed point type (e.g. 0xb is 11, but it could also be 1.1).
+			// That's why we always enforce reformatting fixed point types here.
+			parameter.abiType.type == ABIType::UnsignedFixedPoint ||
+			parameter.abiType.type == ABIType::SignedFixedPoint)
```

Comment:
```suggestion
		if (
			!parameter.matchesBytes(byteRange) ||
			// Fixed point values may look exactly the same, but their actual values may be different if
			// it uses a different fixed point type (e.g. 0xb is 11, but it could also be 1.1).
			// That's why we always enforce reformatting fixed point types here.
			parameter.abiType.type == ABIType::UnsignedFixedPoint ||
			parameter.abiType.type == ABIType::SignedFixedPoint
		)
```

Reply:
Can we just enforce reformatting everywhere?

Reply:
I think we should not touch this in the context of this PR. E.g. https://github.com/ethereum/solidity/pull/10612 did show that some of the automated reformatting created sometimes more confusing outputs. We would need to specify exactly what formatting should be used when. I guess these rules may become quite complicated. If we want this, we should do this in another PR.

Reply:
I agree, then please let us not introduce such a mechanism at all in this PR. Let's either validate everything or nothing.

Reply:
Actually I would prefer not to have such complicated rules. If the formatting is weird, then it is weird. In the end, the tests do not rely on complicated outputs to look nice. If we want it to look nice, we can always change the actual values.

Reply:
Lets say we have an expectation like 
```
// s128x2(fixed128x2): -2.34 -> -2.345
```
Here `-2.345` is wrong, it need to be `-2.34`. We need to enforce the reformatting here, so that `-2.345` will be replace with the correct value `-2.34`. If we don't enforce the reformatting, it will just use `parameter.rawString`, but this parameter was parsed as `-2.345`. So nothing will get replaced.

> I agree, then please let us not introduce such a mechanism at all in this PR. Let's either validate everything or 
> nothing.

Not sure what you mean. We need to enforce the reformatting for fixed point types.

Reply:
I think we should just enforce the reformatting for fixed point types for now. We could improve the formatting rules in another PR.

Reply:
Please do not enforce the reformatting for fixed points. We need a bare minimum of functionality in order to test the changes that have been in these pending pull requests for the past three weeks. Making it look nice is the next step, and we should enforce the reformatting for all types, not as a hack only for fixed point types. And yes, I know, it is not only about looking nice, the formatting can be deceiving, but at least I myself never manually write outputs anyway, so this is not really a trap for me.

Reply:
After short discussion on the chat we decided that it is not that important to be able to support the update, if the `.` was misplaced. I remove this special case for reformatting fixed point types.